Documentation/upgrade.md

```diff
@@ -131,350 +181,165 @@ Any pod that is using a Rook volume should also remain healthy:
+```
+
+
+## Ceph Upgrades
+By default, an upgraded Rook cluster will be running Ceph Luminous (v12), the same Ceph version
+released with Rook v0.8. Now that the Rook and Ceph versions are independently controllable, you can
+choose to update Ceph at any time.
+
+### Ceph images
+Official Ceph container images can be found on [Docker Hub](https://hub.docker.com/r/ceph/ceph/tags/).
+These images are tagged in a few ways:
+* The most explicit form of tags are full-ceph-version-and-build tags (e.g., `v13.2.2-20181023`).
+  These tags are recommended for production clusters, as there is no possibility for the cluster to
+  be heterogeneous with respect to the version of Ceph running in containers.
+* Ceph minor version tags (e.g. `v13.2`) may be desired when a cluster is non-production but where
+  Ceph changes have a possibility of negatively impacting things.
+* Ceph major version tags (e.g., `v13`) are useful for development and test clusters so that the
+  latest version of Ceph is always available.
+
+#### Create your own Ceph image
+**Please first note that Rook issues that cannot be reproduced using the official ceph images cannot
+be supported.** If you wish to create your own Ceph images compatible with Rook, it is fairly
+straightforward. The Ceph image should contain all Ceph binaries built as part of Ceph and whatever
+dependencies are needed for them to work in the container. Installing Ceph RPM or DEB packages in
+the container will likely be the easiest way to achieve this.
+
+### Example upgrade to Ceph Mimic
+The majority of the upgrade will be handled by the Rook operator. Begin the upgrade by changing the
+Ceph image field in the cluster CRD (`spec:cephVersion:image`), then updating the cluster resource.j
+```sh
+export NEW_CEPH_IMAGE='ceph/ceph:v13.2.2-20181023'
+sed -i.bak "s%image: .*%image: $NEW_CEPH_IMAGE%" cluster.yaml
+kubectl -n $ROOK_SYSTEM_NAMESPACE replace -f cluster.yaml
+```
+
+In Rook v0.9.0, the Ceph images for the monitor daemons need updated explicitly. The mgr daemon will
+be the first to update after the above command and will be in `Error` state until the mons are
+updated. This example assumes 3 mons; your cluster configuration may differ. Wait for each mon pod
+to come up `Ready` before proceeding to replace the images for the next mon.
+```sh
+# TODO: Below is not working
+kubectl -n $ROOK_NAMESPACE patch deploy/rook-ceph-mon0 --patch \
+  {"spec": {"template": {"spec": \
+    {"initContainers": [{"name": "monmap-init","image": "'$NEW_CEPH_IMAGE'"}, \
+                          {"name": "mon-fs-init","image": "'$NEW_CEPH_IMAGE'"}]}, \
+    {"containers": [{"name": "mon","image": "'$NEW_CEPH_IMAGE'"}]}}}}
+# wait for Ready
+```
+
+Update dashboard external service HTTP with HTTPS.
+
+### Upgrade Rook and Ceph simultaneously
+When upgrading Rook from v0.8 to v0.9, it is possible to upgrade Ceph simultaneously by changing the
```

Comment:
After #2326 is merged with the v1 crd conversion, I believe this will still work. Let's just make a note to test it after the merge. We will have to update the v1beta1 version of the crd instead of the v1 version that will be found in the release-0.9 branch. Thus, we'll need an additional step to the example above to update the ceph version.